interface/src/app/gene/network/network.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.gene.network', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap',
+  'rzModule'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('gene_network', {
+    url: '/gene_network/?genes&mlmodel',
+    views: {
+      main: {
+        templateUrl: 'gene/network/network.tpl.html',
+        controller: 'GeneNetworkCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Gene Network'}
+  });
+})
+
+.factory('EdgeService', ['$resource', function($resource) {
+  return $resource('/api/v0/edge/');
+}])
+
+.factory('NodeService', ['$resource', function($resource) {
+  return $resource('/api/v0/node/');
+}])
+
+.controller('GeneNetworkCtrl',
+   ['$stateParams', 'EdgeService', 'NodeService', '$log',
+    function GeneNetworkController($stateParams, EdgeService, NodeService,
+                                   $log) {
+      var rawMinWeight = -1.0;
+      var rawMaxWeight = 1.0;
+      var minCorrelation = rawMinWeight;
+      var scaledMaxWeight = rawMaxWeight - rawMinWeight;
+      var setEdgeColor = d3.scale.linear()
+          .domain([0, scaledMaxWeight / 2.0, scaledMaxWeight])
+          .range(['green', 'orange', 'red']);
+
+      var network = d3.network()  // Initialize the network.
+          .minEdge(0)
+          .maxEdge(rawMaxWeight - rawMinWeight)
+          .edgeColor(setEdgeColor)
+          .geneText(function(d) {
+            return d.label;
+          })
+          .legendStart(rawMinWeight)
+          .legendEnd(rawMaxWeight)
+          .legendText('Correlation');
+
+      var self = this;
+      // The following properties of "self" will be available to HTML.
```

Comment:
This is a new Angular feature called `controllerAs` syntax. (I think it was introduced since angular 1.2). As far as I understand, it has two advantages compared to $scope:
(1) It removes $scope dependency.
(2) The HTML code will use **controllerTag.variableName** (**controllerTag** is defined on line 18) to access the properties of **self**, it shows explicitly that **variableName** is from the controller, thus the code is easier to understand.